### PR TITLE
[NCL-7812] Upgrade BrowserRouter to createBrowserRouter (v6.4)

### PR DESCRIPTION
**Pull request by jomrazek**

### CODE REVIEW

**Comment by matedo1** [comment]
Hi @patrikk0123 , for now we agreed `createRoutesFromElements` can be used so that we can keep current declarative structure. In the future we can consider migrating it to object format. Or do you have any preferences?

File: src/AppRoutes.tsx [hunk 1]
```diff
@@ -60,8 +61,8 @@ import { VariablesPage } from 'components/VariablesPage/VariablesPage';
 
 import { AUTH_ROLE } from 'services/keycloakService';
 
-export const AppRoutes = () => (
-  <Routes>
+export const AppRoutes = createRoutesFromElements(
```
Comment (patrikk0123) [inline src/AppRoutes.tsx:64]: I would rather apply the function where the `createBrowserRouter` is called.
This way, AppRoutes remains a component.

**Comment by patrikk0123** [comment]
> Hi @patrikk0123 , for now we agreed `createRoutesFromElements` can be used so that we can keep current declarative structure. In the future we can consider migrating it to object format. Or do you have any preferences?

Using `createRoutesFromElements` is the right choice. The declarative syntax is better and functionally, both ways seem equivalent.

File: src/AppRoutes.tsx (hunk 1 above)
Comment (jomrazek) [reply src/AppRoutes.tsx]: sure :+1:

**Review by patrikk0123 — APPROVED** [review]

**Pull request merged**